Update CONTRIBUTING.rst and CONTRIBUTORS_QUICK_START.rst files with jq installation instructions

nijanthanvijayakumar opened the pull request:
### What files have been modified?
1. CONTRIBUTING.RST
2. CONTRIBUTORS_QUICK_START.rst

### What changes will be made as part of this PR or how does this PR help?
This pull request updates the above mentioned two documentation files by adding the installation information for the `jq` library and marks it as a prerequisite for installing `breeze`. After moving to the ghrc.io images, users are facing issue in setting up breeze and the error that comes up while setting up breeze is as follows:
```
airflow/scripts/ci/libraries/_build_images.sh: line 427: jq: command not found
``` 
Please refer to the below screenshot for the same error message.
![image](https://user-images.githubusercontent.com/6072170/126025000-7694fcfd-36b6-4dab-8e2f-a6a2cba3dd3a.png)

Installing `jq` library on the local machine fixes this issue. This can be done using `sudo apt install jq`. Below screenshot is a proof that the command works on Ubuntu 20.04 machine and installs `jq`.

![IMG_2363](https://user-images.githubusercontent.com/6072170/126025045-c4dd7697-347e-4cf8-91ab-183b74a9ada4.jpeg)

I have run the pre-commits for these changes using `pre-commit run` and it has finished successfully allowing me to commit the changes to the remote branch.

Commit:
Add jq installation info as a prerequisite for setting up breeze

uranusjr commented on CONTRIBUTING.rst:
```diff
@@ -208,27 +208,33 @@ You can configure the Docker-based Breeze development environment as follows:
+
+.. code-block:: bash
+
+  sudo apt install jq
```
We don’t limit this guide to Ubuntu (or Debian) users, so this should say something like

```rst
Install jq on your machine. The exact command depends on the operating system
(or Linux distribution) you use. For example, on Ubuntu:

.. code-block:: bash

    sudo apt install jq

Or on macOS with Homebrew:

.. code-block:: bash

    brew install jq
```

nijanthan-dev replied:
Thanks for the review & prompt response @uranusjr .

I am a Mac user too and I used the `brew install jq` command to install that library. However, if you look at this `CONTRIBUTORS_QUICK_START.rst` document, the section that this change comes under uses the same Ubuntu command patterns. Hence, I have limited it to Ubuntu, to keep it consistent with the other prerequisites. My two cents is, there should be a separate section with the prerequisite for Mac OS and this should go in there. Let me know what you think?

potiuk replied:
For quick-start yeah. But the `CONTRIBUTING.rst` change is more "generic". I think this CONTRIBUTING is for people who know what they are doing, know what system they have, and they can follow if ubuntu - this if os - that. So what @uranusjr proposes is good. Those examples here is more what we say "examples" - and for people who are experts, they might want to have other ways of installing those deps.

The `Quick-start` is more for people who wanted to quickly go through the steps, sometimes without full knowledge of what they are doing (but willing to learn it) and there it is better to have two chapters "prerequisites for Ubuntu" and "prerequisites for Mac" with more detailed and accurate instructions and troubleshooting. These should be much more "robust" and guide the user step-by-step.

We miss the 'MacOS' part there - another PR to do :)

nijanthan-dev replied:
Great discussion. Thanks, @potiuk and @uranusjr, for pointing out. I have updated the `CONTRIBUTING.rst` file with the example for the macOS as well, as pointed outed by @uranusjr.

The pre-commit and the GitHub checks have passed. Regarding adding the macOS prerequisites to the `CONTRIBUTORS_QUICK_START.rst`, I will send over a new PR with more robust examples for macOS. I hope my changes are looking good now. 

Would you please let me know if any more comments? I am happy to learn and address them

nijanthanvijayakumar committed:
Add macOS example for jq installation using Homebrew

The "Configure Your Environment" section was missing the examples on
installing jq on macOS. Earlier version of the document only had the
example for the Ubuntu OS.
